File: dotnet-desktop-guide/net/winforms/printing/snippets/how-to-print-text-document/csharp/Form1.cs
namespace Sample_Print_Application
{
    public partial class Form1 : Form
    {
        //<add_string_ to_your_form>
        private string stringToPrint;
        //</add_string_ to_your_form>

        public Form1()
        {
            InitializeComponent();
        }
        private void ReadFile()
        {
            //<set_DocumentName_then_open_and_read_document_to_the_string>
            string docName = "testPage.txt";
            string docPath = @"C:\";
            printDocument1.DocumentName = docName;
            using (FileStream stream = new FileStream(docPath + docName, FileMode.Open))
            using (StreamReader reader = new StreamReader(stream))
            {
                stringToPrint = reader.ReadToEnd();
            }
            //</set_DocumentName_then_open_and_read_document_to_the_string>
        }

        //<print_contents_of_the_file_using_event_handler>
        private void PrintDocument1_PrintPage(object sender, System.Drawing.Printing.PrintPageEventArgs e)
        {
            int charactersOnPage = 0;
            int linesPerPage = 0;

            // Sets the value of charactersOnPage to the number of characters
            // of stringToPrint that will fit within the bounds of the page.
            e.Graphics.MeasureString(stringToPrint, this.Font,
                e.MarginBounds.Size, StringFormat.GenericTypographic,
                out charactersOnPage, out linesPerPage);

            // Draws the string within the bounds of the page
            e.Graphics.DrawString(stringToPrint, this.Font, Brushes.Black,
                e.MarginBounds, StringFormat.GenericTypographic);

            // Remove the portion of the string that has been printed.
            stringToPrint = stringToPrint.Substring(charactersOnPage);

            // Check to see if more pages are to be printed.
            e.HasMorePages = (stringToPrint.Length > 0);
        }
        //</print_contents_of_the_file_using_event_handler>
        
        private void button1_Click(object sender, EventArgs e)
        {
            ReadFile();
            //<call_print_method_to_print_file>
            printDocument1.Print();
            //</call_print_method_to_print_file>
        }
    }
}

File: dotnet-desktop-guide/net/winforms/printing/snippets/how-to-print-text-document/vb/Form1.vb
Imports System.IO
Imports System.Drawing
Imports System.Drawing.Printing
Imports System.Windows.Forms

Public Class Form1

    '<add_string_to_your_form>
    'Private PrintDocument1 As New PrintDocument()
    Private stringToPrint As String
    '</add_string_to_your_form>

    '<print_contents_using_event_handler>
    Private Sub PrintDocument1_PrintPage(ByVal sender As Object,
    ByVal e As PrintPageEventArgs) Handles PrintDocument1.PrintPage

        Dim charactersOnPage As Integer = 0
        Dim linesPerPage As Integer = 0

        ' Sets the value of charactersOnPage to the number of characters 
        ' of stringToPrint that will fit within the bounds of the page.
        e.Graphics.MeasureString(stringToPrint, Me.Font, e.MarginBounds.Size,
        StringFormat.GenericTypographic, charactersOnPage, linesPerPage)

        ' Draws the string within the bounds of the page
        e.Graphics.DrawString(stringToPrint, Me.Font, Brushes.Black,
        e.MarginBounds, StringFormat.GenericTypographic)

        ' Remove the portion of the string that has been printed.
        stringToPrint = stringToPrint.Substring(charactersOnPage)

        ' Check to see if more pages are to be printed.
        e.HasMorePages = stringToPrint.Length > 0

    End Sub
    '</print_contents_using_event_handler>

    '<set_DocumentName_and_string>
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click

        Dim docName As String = "testPage.txt"
        Dim docPath As String = "C:\"
        Dim fullPath As String = System.IO.Path.Combine(docPath, docName)

        PrintDocument1.DocumentName = docName

        stringToPrint = System.IO.File.ReadAllText(fullPath)
        
        '<call_print_method_to_print_file>
        PrintDocument1.Print()
        '</call_print_method_to_print_file>

    End Sub
    '</set_DocumentName_and_string>

End Class
